### PR TITLE
Add shadow inconsistency opacity fix with updated tests

### PR DESCRIPTION
**Pull request by lyzhan7**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

Planning to make the same change upstream into RN core

## Summary

Revisiting https://github.com/microsoft/react-native-macos/commit/a598ee4eff9364f5e8865de092f7d368388eb032 - this got reverted in https://github.com/microsoft/react-native-macos/pull/1385 because it introduced an issue with a test. After more investigation, it seems like the test was incorrect and the error was expected. This PR is to add back the original changes + fix an ifdef bug that was in the original change + update the test.

The issue with the test is that the react native Text component has its own props for shadows - i.e. [textShadowColor, textShadowRadius, etc](https://reactnative.dev/docs/text-style-props) and the shadow props that View uses does not work on Text (i.e. shadowColor, shadowRadius, etc.). Not entirely sure how this test was passing without any errors before my change, it seems like something about the change is causing (expected) errors to actually show up.

## Changelog

[macOS] [Fixed] - Shadow consistency opacity fix with test updates

## Test Plan

Confirmed that on iOS and macOS, setting shadowColor/shadowOffset/etc. does not work on Text, only textShadowColor/textShadowOffset/etc works. Confirmed that after updating the tests, the tests pass.

### CODE REVIEW

File: packages/rn-tester/js/examples/AnimatedGratuitousApp/AnExSet.js [hunk 1]
```diff
@@ -139,10 +139,10 @@ const styles = StyleSheet.create({
...
     fontSize: 25,
     color: 'white',
-    shadowRadius: 3,
-    shadowColor: 'black',
-    shadowOpacity: 1,
-    shadowOffset: {height: 1},
+    textShadowRadius: 3,
+    textShadowColor: 'black',
+    textShadowOpacity: 1,
+    textShadowOffset: {height: 1},
```
Comment (Saadnajmi) [inline packages/rn-tester/js/examples/AnimatedGratuitousApp/AnExSet.js:145]: Can you mark this with `// TODO(macOS tags)` and also reference a Github issue to upstream this fix, for tracking purposes?

**Review by Saadnajmi — APPROVED** [review]

**Review by chiuam — APPROVED** [review]

**Review by mischreiber — APPROVED** [review]

**Comment by lyzhan7** [comment]
Turns out textShadowOpacity does not exist (even though shadowOpacity does) - wasn't sure if this was necessary but went ahead and made sure the opacity was set to 1 in the textShadowColor. Also don't really know why it wasn't complaining before about omitting width...

File: packages/rn-tester/js/examples/AnimatedGratuitousApp/AnExSet.js [hunk 2]
```diff
@@ -139,10 +139,10 @@ const styles = StyleSheet.create({
   headerText: {
     fontSize: 25,
     color: 'white',
-    shadowRadius: 3,
-    shadowColor: 'black',
-    shadowOpacity: 1,
-    shadowOffset: {height: 1},
+    // TODO(macOS 1409)
```
Comment (Saadnajmi) [inline packages/rn-tester/js/examples/AnimatedGratuitousApp/AnExSet.js:142]: Missing GH#

Comment (Saadnajmi) [reply packages/rn-tester/js/examples/AnimatedGratuitousApp/AnExSet.js:142]: And also [ and and ending tag with ]

**Pull request merged**